src/keys.rs

```diff
@@ -134,6 +134,21 @@ impl<'a, T: PrimaryKey<'a> + Prefixer<'a> + KeyDeserialize, U: PrimaryKey<'a> +
     }
 }
 
+// implements PrimaryKey for all &T where T implements PrimaryKey.
+impl<'a, T> PrimaryKey<'a> for &'a T
+where
+    T: PrimaryKey<'a>,
+{
+    type Prefix = <T as PrimaryKey<'a>>::Prefix;
+    type SubPrefix = <T as PrimaryKey<'a>>::SubPrefix;
+    type Suffix = <T as PrimaryKey<'a>>::Suffix;
+    type SuperSuffix = <T as PrimaryKey<'a>>::SuperSuffix;
```

Comment:
nit: I don't think the type casts are needed, `T::Prefix` should work equally well, since `T` is unambiguously PrimaryKey.

Reply:
nice! thank you.